src/test/java/core/basesyntax/service/impl/FruitShopDataProcessingServiceTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataProcessingService;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.operation.BalanceOperationHandler;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.PurchaseOperationHandler;
+import core.basesyntax.service.operation.ReturnOperationHandler;
+import core.basesyntax.service.operation.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitShopDataProcessingServiceTest {
+    private static final String DEFAULT_FRUIT_NAME = "banana";
+    private static final Map<Operation, OperationHandler> operationHandlerMap = Map.of(
+            Operation.BALANCE, new BalanceOperationHandler(),
+            Operation.SUPPLY, new SupplyOperationHandler(),
+            Operation.PURCHASE, new PurchaseOperationHandler(),
+            Operation.RETURN, new ReturnOperationHandler());
+    private OperationStrategy operationStrategy;
+    private DataProcessingService dataProcessingService;
+
+    @BeforeEach
+    void setUp() {
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        dataProcessingService = new FruitShopDataProcessingService(operationStrategy);
+    }
+
+    @Test
+    void processData_validFruitTransactions_ok() {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        fruitTransactions.add(new FruitTransaction(DEFAULT_FRUIT_NAME, 20, Operation.BALANCE));
```

Comment:
This could be added to setUp method or create method presetWithData and move this fruitTransactions.add there